packages/itp-rcc-button/src/Button.js

```diff
@@ -0,0 +1,178 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+function insertSpace(child) {
+  if (child == null) {
+    return
+  }
+
+  if (typeof child === 'string') {
+    return <span>{child}</span> // eslint-disable-line consistent-return
+  }
+
+  return child // eslint-disable-line consistent-return
+}
+
+export default class Button extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      loading: props.loading,
+    }
+
+    this.handleClick.bind(this)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const currentLoading = this.props.loading
+    const loading = nextProps.loading
+
+    if (currentLoading) {
+      clearTimeout(this.delayTimeout)
+    }
+
+    if (typeof loading !== 'boolean' && loading && loading.delay) {
+      this.delayTimeout = window.setTimeout(
+        () => this.setState({loading}),
+        loading.delay,
+      )
+    } else {
+      this.setState({loading})
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.delayTimeout) {
+      clearTimeout(this.delayTimeout)
+    }
+  }
+
+  handleClick(e) {
```

Comment:
arrow function instead of binding in constructor

Reply:
this is strange, as in daikin we use it all over without overwriting the defaults.. could you for now overwrite those rules in the eslintrc in the project? Then idd, we can check to change the defaults inside itp-react-script if necessary

Reply:
client project was at `^0.44.2`, comes with `babel-eslint`, updating `package.json`...